contracts/protocols/mainnet/compound_iii/helpers.sol

```diff
@@ -0,0 +1,433 @@
+        uint64 priceScale;
+        ///@dev The scale for the index tracking protocol rewards, useful in calculating rewards APR
+        uint64 trackingIndexScale;
+    }
+
+    struct Token {
+        uint8 offset;
+        uint8 decimals;
+        address token;
+        string symbol;
+        ///@dev 10**decimals
+        uint256 scale;
+    }
+
+    struct AssetData {
+        Token token;
+        ///@dev token's priceFeed
+        address priceFeed;
+        ///@dev answer as per latestRoundData from the priceFeed scaled by priceScale
+        uint256 price;
+        ///@dev 10**decimals
+        uint64 scale;
+        ///@dev The collateral factor(decides how much each collateral can increase borrowing capacity of user),
+        //integer representing the decimal value scaled up by 10 ^ 18.
+        uint64 borrowCollateralFactor;
+        ///@dev sets the limits for account's borrow balance,
+        //integer representing the decimal value scaled up by 10 ^ 18.
+        uint64 liquidateCollateralFactor;
+        ///@dev liquidation penalty deducted from account's balance upon absorption
+        uint64 liquidationFactor;
+        ///@dev integer scaled up by 10 ^ decimals
+        uint128 supplyCap;
+        ///@dev  current amount of collateral that all accounts have supplied
+        uint128 totalCollateral;
+    }
+
+    struct AccountFlags {
+        ///@dev flag indicating whether the user's position is liquidatable
+        bool isLiquidatable;
+        ///@dev flag indicating whether an account has enough collateral to borrow
+        bool isBorrowCollateralized;
+    }
+
+    struct MarketFlags {
+        bool isAbsorbPaused;
+        bool isBuyPaused;
+        bool isSupplyPaused;
+        bool isTransferPaused;
+        bool isWithdrawPaused;
+    }
+
+    struct UserCollateralData {
+        Token token;
+        ///@dev current positive base balance of an account or zero
+        uint256 suppliedBalanceInBase;
+        uint256 suppliedBalanceInAsset;
+    }
+
+    struct RewardsConfig {
+        address token;
+        uint64 rescaleFactor;
+        bool shouldUpScale;
+        ///@dev The minimum amount of base principal wei for rewards to accrue. The minimum amount of base asset supplied to the protocol in order for accounts to accrue rewards.
+        uint104 baseMinForRewards;
+    }
+
+    struct UserRewardsData {
+        address rewardToken;
+        uint8 rewardTokenDecimals;
+        uint256 amountOwed;
+        uint256 amountClaimed;
+    }
+
+    struct UserData {
+        int104 baseBalance;
+        ///@dev the base balance of supplies with interest, 0 for borrowing case or no supplies
+        uint256 suppliedBalanceInBase;
+        ///@dev the borrow base balance including interest, for non-negative base asset balance value is 0
+        uint256 borrowedBalanceInBase;
+        ///@dev
+        uint16 assetsIn;
+        uint64 accountTrackingIndex;
+        uint64 interestAccrued;
```

Comment:
@Richa-iitr `interestAccrued` on base supplied amount and `suppliedBalanceInBase` doesn't include `interestAccrued`, right?

Reply:
Yes `interestAccrued` is interest accrued till now by the account on his supplied base asset. 
`suppliedBalanceInBase` includes the interest.

Reply:
okay! So, `interestAccrued` is extra information. And we can just rely on `suppliedBalanceInBase`

Reply:
yes